src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
+		boolean result2 = checkIINMaskCorrectness (number, masks);
+		boolean result3 = checkLuhnalgorithmCorrectness(number);
+		
+		return result1 && result2 && result3;
+	}
+	
+	private boolean checkLengthCorrectness(String number, ArrayList<Integer> chars){
+		boolean result = false;
+		for (int i = 0; i<chars.size();i++){
+			if (chars.get(i) == number.length())
+				result = true;
+		}
+		return result;
+	}
+	
+	private boolean checkIINMaskCorrectness (String number, ArrayList<String> masks){
+		boolean result = false;
+		for (int i = 0; i<masks.size();i++){
+			if (number.startsWith(masks.get(i)))
+				result = true;
+		}
```

Comment:
Iterating over masks and checking if number as String begins with it wasn't probably the intention : - )

Original intention was to validate if each part of a card number matches given range/value:
- category of entity which issued the card
- issuer identification number
- account number
- checksum

For that reason would be better to prepare better **specific model** for masks 
+ rewrite this method and encapsulate it in Vendor class